instaloader/structures.py

```diff
@@ -1175,13 +1186,17 @@ def get_posts(self) -> NodeIterator[Post]:
         self._obtain_metadata()
         return NodeIterator(
             self._context,
-            '003056d32c2554def87228bc3fd9668a',
-            lambda d: d['data']['user']['edge_owner_to_timeline_media'],
-            lambda n: Post(self._context, n, self),
-            {'id': self.userid},
+            '',
```

Comment:
Passing an empty string as query_hash and keeping an empty string in NodeIterator is not so nice I think. I suggest to make the argument an `Optional[str]` in the NodeIterator constructor and passing all parameters by their name in this invocation. Due to the position of the parameter we cannot entirely leave out the query_hash parameter, but I think verbosely passing `query_hash=None` rather than having an empty string at the parameter's position makes this invocation easier to understand.